sources/api/netdog/src/net_config/v3.rs

```diff
@@ -0,0 +1,106 @@
+//! The `v3` module contains the third version of the network configuration and implements the
+//! appropriate traits.
+
+use super::devices::NetworkDeviceV1;
+use super::{error, Interfaces, Result, Validate};
+use crate::interface_name::InterfaceName;
+use crate::wicked::{WickedInterface, WickedLinkConfig};
+use indexmap::IndexMap;
+use serde::Deserialize;
+use snafu::ensure;
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct NetConfigV3 {
+    #[serde(flatten)]
+    pub(crate) net_devices: IndexMap<InterfaceName, NetworkDeviceV1>,
+}
+
+impl Interfaces for NetConfigV3 {
+    fn primary_interface(&self) -> Option<String> {
+        self.net_devices
+            .iter()
+            .find(|(_, v)| v.primary() == Some(true))
+            .or_else(|| self.net_devices.first())
+            .map(|(n, _)| n.to_string())
+    }
+
+    fn has_interfaces(&self) -> bool {
+        !self.net_devices.is_empty()
+    }
+
+    fn as_wicked_interfaces(&self) -> Vec<WickedInterface> {
+        let mut wicked_interfaces = Vec::new();
+        for (name, config) in &self.net_devices {
+            let interface = WickedInterface::from((name, config));
+
+            // If config is a Bond, we will generate the interface configuration. For each interface:
+            // WickedInterface::new(name), add that to wicked_interfaces
+            if let NetworkDeviceV1::BondDevice(b) = config {
+                for device in &b.interfaces {
+                    let mut wicked_sub_interface = WickedInterface::new(device.clone());
+                    wicked_sub_interface.link = Some(WickedLinkConfig {
+                        master: name.clone(),
+                    });
+
+                    wicked_interfaces.push(wicked_sub_interface)
+                }
+            }
+
+            wicked_interfaces.push(interface)
+        }
+
+        wicked_interfaces
+    }
+}
+
+#[allow(clippy::to_string_in_format_args)]
+impl Validate for NetConfigV3 {
```

Comment:
I went back and forth on whether to do this. I was attempting to avoid touching more code than strictly necessary to achieve this new functionality but I can refactor this out. I'll add the changes as a separate commit for it though because its now getting into refactoring and cleaning up the code base vs simply adding in the needed code for bonding and vlans.

Reply:
I've moved the v2 validation to the new style but v1 didn't have static addressing which is where most of this actually came in. So it is unchanged.